resotocore/tests/resotocore/core_config_test.py

```diff
@@ -92,6 +92,46 @@ def test_validate() -> None:
     }
 
 
+def test_config_override(config_json: Json) -> None:
+    # config_json is a valid parsable config
+    cfg = deepcopy(config_json)
+
+    # create a temp file with a custom config
+    with TemporaryDirectory() as tmp:
+        # config with env var override
+        hosts_conf = Path(tmp, "foobar.yml")
+        hosts_conf.write_text(
+            """
+api:
+    web_hosts: ["$(WEB_HOST)"]
+        """,
+            encoding="utf-8",
+        )
+
+        # config that overrides the default config and containes an env var which can't be resolved
+        other_conf = Path(tmp, "config.yml")
+        other_conf.write_text(
+            """
+api:
+    web_port: 1337
```

Comment:
Please create a test where you replace an int (like the port) via env vars.
I wonder if the type checker will complain and needs special handling?